File: apps/container_migration/backend/models/config_model.py
from pydantic import BaseModel
from typing import List, Optional

class RuleConfig(BaseModel):
    rule: str
    cluster: str
    action: str
    targetCluster: Optional[str] = None
    forensic_analysis: Optional[bool] = False
    AI_suggestion: Optional[bool] = False
    registry_address: Optional[str] = None
    disable_istio_sidecar: Optional[bool] = False
    skip_cpu_compat_check: Optional[bool] = False
    cleanup_incompatible_mounts: Optional[bool] = False

class Config(BaseModel):
    config: List[RuleConfig]

File: apps/container_migration/frontend/src/app/model/config.model.ts
export interface RuleConfig {
    rule: string;
    cluster: string;
    action: string;
    targetCluster?: string | null;
    forensic_analysis: boolean;
    AI_suggestion: boolean;
    registry_address?: string | null;
    istio_routing_context?: string | null;
    disable_istio_sidecar?: boolean;
    skip_cpu_compat_check?: boolean;
    cleanup_incompatible_mounts?: boolean;
}

export interface Config {
    config: RuleConfig[];
}
